scene/gui/spin_box.cpp

```diff
@@ -83,7 +83,7 @@ void SpinBox::_text_submitted(const String &p_string) {
 
 		err = expr->parse(text);
 		if (err != OK) {
-			return;
+			expr->parse(last_updated_text);
```

Comment:
```suggestion
			err = expr->parse(last_updated_text);
		}
		if (err != OK) {
			return;
		}
```
Or things will break

Reply:
The value is initialized as 0 as part of the range class it inherits, but I agree it's better safe than sorry.